fbw-common/src/wasm/systems/systems/src/structural_flex/wing_flex.rs

```diff
@@ -0,0 +1,2089 @@
+            aileron_left_2_position_id: context
+                .get_identifier("HYD_AIL_LEFT_MIDDLE_DEFLECTION".to_owned()),
+            aileron_left_3_position_id: context
+                .get_identifier("HYD_AIL_LEFT_OUTWARD_DEFLECTION".to_owned()),
+
+            aileron_right_1_position_id: context
+                .get_identifier("HYD_AIL_RIGHT_INWARD_DEFLECTION".to_owned()),
+            aileron_right_2_position_id: context
+                .get_identifier("HYD_AIL_RIGHT_MIDDLE_DEFLECTION".to_owned()),
+            aileron_right_3_position_id: context
+                .get_identifier("HYD_AIL_RIGHT_OUTWARD_DEFLECTION".to_owned()),
+
+            flaps_left_position_id: context
+                .get_identifier("LEFT_FLAPS_POSITION_PERCENT".to_owned()),
+            flaps_right_position_id: context
+                .get_identifier("RIGHT_FLAPS_POSITION_PERCENT".to_owned()),
+            lateral_offset: Ratio::default(),
+
+            spoilers_left_position: [0.; 8],
+            spoilers_right_position: [0.; 8],
+
+            ailerons_left_position: [0.5; 3],
+            ailerons_right_position: [0.5; 3],
+
+            left_flaps_position: Ratio::default(),
+            right_flaps_position: Ratio::default(),
+
+            left_wing_lift: Force::default(),
+            right_wing_lift: Force::default(),
+
+            standard_lift_spread: Vector5::from(Self::NOMINAL_WING_LIFT_SPREAD_RATIOS),
+            lift_left_table_newton: Vector5::default(),
+            lift_right_table_newton: Vector5::default(),
+        }
+    }
+
+    fn update(&mut self, total_lift: Force) {
+        self.compute_lift_modifiers(total_lift);
+    }
+
+    fn compute_lift_modifiers(&mut self, total_lift: Force) {
+        let wing_base_left_spoilers =
+            (self.spoilers_left_position[0] + self.spoilers_left_position[1]) / 2.;
+        let wing_mid_left_spoilers = (self.spoilers_left_position[2]
+            + self.spoilers_left_position[3]
+            + self.spoilers_left_position[4]
+            + self.spoilers_left_position[5]
+            + self.spoilers_left_position[6]
+            + self.spoilers_left_position[7])
+            / 6.;
+
+        let wing_base_right_spoilers =
+            (self.spoilers_right_position[0] + self.spoilers_right_position[1]) / 2.;
+        let wing_mid_right_spoilers = (self.spoilers_right_position[2]
+            + self.spoilers_right_position[3]
+            + self.spoilers_right_position[4]
+            + self.spoilers_right_position[5]
+            + self.spoilers_right_position[6]
+            + self.spoilers_right_position[7])
+            / 6.;
+
+        let left_ailerons_mid = ((self.ailerons_left_position[0] - 0.5) * 2.
+            + (self.ailerons_left_position[1] - 0.5) * 2.)
+            / 2.;
+        let right_ailerons_mid = ((self.ailerons_right_position[0] - 0.5) * 2.
+            + (self.ailerons_right_position[1] - 0.5) * 2.)
+            / 2.;
```

Comment:
Problem with those is in the end I don't understand what it's doing while with "too much" multiplications compiler would probably end up with the same machine code :p

Reply:
Actually the compiler will not optimize the unecessary multiplications away because of floating point semantics. What I suggest instead is to write your formula as a comment before the statement.